### DISCUSSION TITLE
Is IIGS Support remotely possible?

### DISCUSSION THREAD

**Discussion by Alex-Kw** [markadev/AppleII-VGA#22] [Ideas]
I've had several folks ask me if IIGS video mode(s) support is possible. I'm unsure if it would be possible to fully support it. I would love to hear ideas even if it's just why it cannot be done or what hurdles would have to be overcome.

**Comment by markadev** [comment]
Hey there, I'm not too familiar with the IIgs bus but it looks like it has a couple extra bus signals that would need to be snooped and the bus supports multiple speeds, which would need to be adjusted for.  The new IIgs modes also support more bits-per-pixel for color so the current 3x3-bit DAC output would not look very good unless it was expanded to 3x4 bits.

The [V2 Analog project](https://www.v2retrocomputing.com/analog), which was based off of this project so also uses a Pico, has some IIgs mode support but also doesn't have the extra bus signals hooked up so they've documented the limitation there:

> Partial compatibility with Apple II GS systems. The ∀2 Analog Rev1 hardware does not have access to the M2B0 or M2SEL signals on the bus to qualify 16/24 bit addresses.  This means any accesses outside the shadow memory banks will probably cause visible glitches in VGA output and may cause the card to incorrectly interpret memory accesses as soft-switch or register manipulation.

Unfortunately the current design is using up almost all of the Pico's GPIO pins so to add 2 extra inputs from the Apple bus and 3 extra VGA DAC output bits would be the challenge.  I won't say it's impossible because I haven't thought about it at all... maybe it could be done with some additional external glue logic.

**Comment by jtflanagan** [comment]
I'm working on another project, which started as something very similar to AppleII-VGA but has diverged significantly at this point. I do still use a Pico, but only for the PIO feature to do bus control.  All the heavy lifting is being done by a Teensy 4.1.  It has a lot more pins, so I'm able to bring in all the bus signals I care about.  My first (breadboarded) prototype which I demoed at Kansasfest 2023 did not support IIgs, but I'm working on a PCB version which will.  I've never done a PCB before so it's a bit slow going, but if you want to see the current progress, you can look at the github project: [https://github.com/jtflanagan/AppleTini](https://github.com/jtflanagan/AppleTini)

I only just pushed my first reasonably-complete schematic and PCB this evening, but they haven't been extensively checked and I would not be surprised if I made a ton of rookie mistakes.  Also the pico and teensy code are out of date, I need to update them to match the pin rearrangement I did while laying out the PCB.

Do note that I'm aiming a lot higher.  AppleTini is intended to, at the base level, replace the VidHD card which provides HDMI-out for all Apple II modes, including IIgs-specific modes.  But I have a lot more features in mind!  Next year's Kansasfest presentation will be very exciting.
